tinygrad/codegen/linearizer.py

```diff
@@ -338,10 +338,7 @@ def end_loop(xx:List[Variable]):
         fake_global_idxs = [x*0 for x in global_idxs]
         self.global_store(-1, fake_global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs, acc)  # store accumulators
         self.uop(UOps.BARRIER, None, [], ())
-        end_loop(loop_local_idxs)
-
-        # local indexs are over, 0 them out
-        local_idxs = [x*0 for x in local_idxs]
+        end_loop([Variable.num(0)]*self.local_dims + loop_local_idxs[self.local_dims:])
```

Comment:
Hmm...I don't think this function does what you think it does

Reply:
Right sorry this was a hack to get the COND_BRANCH generated correctly.